### PR TITLE
Pattern Lab UI Fixes + Polish

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
TBD

## Summary
TLDR: a bunch of misc Pattern Lab UI updates and bug fixes (the original work that https://github.com/pattern-lab/patternlab-node/pull/1094 is based off of).

15237A78-56C4-4476-8CEE-EEEE0E8F8D7F
60721847-8549-4990-A401-8FB6174C491E

These updates include:
- Reduced the # of initial UI re-renders (particularly unnecessary re-renders of the Nav) via updates to the main HTML template + Layout component 
- CSS updates to fix / improve small screen support — in particular, trying to make sure the navigation and viewport content don’t get chopped off!
- Added the ability to auto-close the primary navigation via pressing the escape key + clicking outside the Nav (including slight behavior differences for the mobile panel vs larger screen dropdown / side nav)
- Updated the “Open in a New Tab” UI to improve the consistency of updating when expected 
- Updated the Header UI on smaller screens to now display a Close icon when the Nav is open!
- Updated the Viewport size UI to more reliably display the current viewport size (in particular when initially loading)
- Started cleaning up the Nav component’s internal logic + updated to no longer auto-close the mobile navigation section you are currently on, in addition to no longer “forgetting” which section you are on while toggling between the vertical and horizontal layout modes on larger screens
- Theoretically added / restored support for setting the default theme color / layout mode via the Pattern Lab config (need to further test to verify)

## Updates From https://github.com/pattern-lab/patternlab-node/pull/1094 Already Merged:
- ~Updated the UI for expanding / collapsing in-page code panels + corresponding icon / naming updates used in the Context / Tools menu~
- ~Updated CSS for Pattern Lineage UI to better allow links to wrap if needed~
- ~Fixed placeholder text color contrast issues in the global search input~
- ~Ported over the “sticky” copy to clipboard behavior (copy to clipboard button stays at the top of the code snippet, even while scrolling!).~
- ~Added the Loader UI (spinner) inside Drawer when source code is getting loaded~
- ~Added mobile / small screen support for displaying the Nav as a slide out panel (aka vertical nav) + dropdown to allow for the exact same layout options on all screen sizes~
  - ~This also sets us up for a larger Nav UI refactor now that we don’t have totally different Nav behavior on smaller vs larger screen sizes!~
- ~Polished up the UI for the Viewport resizer per helpful feedback from @mikemai2awesome~
- ~The bug fixes for opening / closing code panels https://github.com/pattern-lab/patternlab-node/pull/1074~
- ~Added Copy / Copied SVG icons for the copy to clipboard button~

### CODE REVIEW

**Comment by sghoweri** [comment]
> It doesn't always happen in a regular window, but it does in incognito, which makes me wonder if there's something stored in cookies or local storage that's not working as expected.
> 
> Can you reproduce this? Want to fix it here or separately?

@danielamorse weird - I can't seem to replicate this issue locally in Chrome in Incognito mode... 🤔 

I'm fine digging further into this separately but I'll let you make the call on this.

**Comment by danielamorse** [comment]
@sghoweri I retested this locally and I can no longer replicate the bug if incognito or not 👍 

I did a full `clean` and `setup` before testing this time. Maybe that made a difference. Maybe user error! Anyway, approved 🙏

**Comment by sghoweri** [comment]
Thanks @danielamorse!!

**Comment by sghoweri** [comment]
PR was released with v2.11.0
